shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -338,6 +338,10 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
         PolymorphicType(MinLevel, fun(singleTup(v), fun(singleTup(v), BoolType)(noProv))(noProv))
       },
       "error" -> BotType,
+      "," -> {
+        val v = freshVar(noProv, N)(1)
+        PolymorphicType(MinLevel, fun(TupleType(N -> TopType.toUpper(provTODO) :: N -> v.toUpper(provTODO) :: Nil)(noProv), v)(noProv))
+      },
```

Comment:
Is this line of code causing so many inferred type changes?

Reply:
Yup. The `freshVar` call, to be specific.

Reply:
All right, I changed it slightly so all the meaningless type changes are gone.

Calls for a better solution in the future, though.